tests/Mado/QueryBundle/Component/Meta/JsonPathFinderTest.php

```diff
@@ -0,0 +1,466 @@
+    {
+        $this->samepleJson = [
+            "AppBundle\\Entity\\Bar" => [
+                "relations" => [
+                    "fizz" => "AppBundle\\Entity\\Fizz",
+                ]
+            ],
+        ];
+
+        $this->mapper = $this
+            ->getMockBuilder('Mado\QueryBundle\Component\Meta\RelationDatamapper')
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->mapper->expects($this->once())
+            ->method('getMap')
+            ->will($this->returnValue(
+                $this->samepleJson
+            ));
+
+        $this->pathFinder = new JsonPathFinder(
+            $this->mapper
+        );
+
+        $this->assertEquals(
+            "fizz",
+            $this->pathFinder->getSourceRelation("AppBundle\\Entity\\Fizz")
+        );
+    }
+
+    public function testLookForPathFromStartEntityToDestination()
+    {
+        $this->samepleJson = [
+            "AppBundle\\Entity\\Root" => [
+                "relations" => [
+                    "foo" => "AppBundle\\Entity\\Foo",
+                ]
+            ],
+            "AppBundle\\Entity\\Foo" => [
+                "relations" => [
+                    "bar" => "AppBundle\\Entity\\Bar",
+                ]
+            ],
+            "AppBundle\\Entity\\Bar" => [
+                "relations" => [
+                    "fizz" => "AppBundle\\Entity\\Fizz",
+                ]
+            ],
+        ];
+
+        $this->mapper = $this
+            ->getMockBuilder('Mado\QueryBundle\Component\Meta\RelationDatamapper')
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->mapper->expects($this->once())
+            ->method('getMap')
+            ->will($this->returnValue(
+                $this->samepleJson
+            ));
+
+        $this->pathFinder = new JsonPathFinder(
+            $this->mapper
+        );
+
+        $this->pathFinder->setEntity("AppBundle\\Entity\\Root");
+
+        $this->assertEquals(
+            "foo.bar.fizz",
+            $this->pathFinder->getPathTo("AppBundle\\Entity\\Fizz")
+        );
+    }
+
+    public function testBuildPathBetweenTwoEntities()
+    {
+        $this->samepleJson = [
+            "GammaBundle\\Entity\\Item" => [
+                "relations" => [
+                    "items" => "AppBundle\\Entity\\Foo",
+                ]
+            ],
+            "AppBundle\\Entity\\Foo" => [
+                "relations" => [
+                    "item" => "ZarroBundle\\Entity\\Item",
+                ]
+            ],
+            "ZarroBundle\\Entity\\Item" => [
+                "relations" => [
+                    "family" => "AppBundle\\Entity\\Family",
+                ]
+            ],
+        ];
+
+        $this->mapper = $this
+            ->getMockBuilder('Mado\QueryBundle\Component\Meta\RelationDatamapper')
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->mapper->expects($this->once())
```

Comment:
you can remove expects method here

Reply:
Why? The method getMap() must be called just once.

Reply:
?